Feat: Add authentik provider (#120)
pilcrowOnPaper/arctic: docs/pages/providers/authentik.md

---
title: "Authentik"
---

# Authentik

For usage, see [OAuth 2.0 provider with PKCE](/guides/oauth2-pkce).

> ```ts

Reviewer: Please update the code block to be consistent with other providers

import { Authentik } from "arctic";

const AUTHENTIK_URL = "http://localhost:9000";

export const authentik = new Authentik(
	AUTHENTIK_URL,
	AUTHENTIK_CLIENT_ID,
	AUTHENTIK_CLIENT_SECRET,
	AUTHENTIK_REDIRECT
);
```

> Authentik with version 2024.2 and higher receive only access token. To get the refresh token as well, you need to include scope `offline_access` in the request. The scope also needs to be enabled in the provider advanced settings.

Reviewer: ```suggestion
Authentik with version 2024.2 and higher only provides the access token by default. To get the refresh token as well, you'll need to include the `offline_access` scope. The scope also needs to be enabled in your app's advanced settings (Application > Providers > Edit > Advanced protocol settings > Scopes).
```

> `Application > Providers > Edit > Advanced protocol settings > Scopes`

Reviewer: ```suggestion
```

```ts
const scopes = ["profile", "email", "openid", "offline_access"];

const url: URL = await authentik.createAuthorizationURL(state, {

> 	scopes

Reviewer: ```suggestion
	scopes: ["profile", "email", "openid", "offline_access"]
```
